cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/NamespaceExistenceVerifier.java

```diff
@@ -19,28 +19,37 @@
 import co.cask.cdap.common.NamespaceNotFoundException;
 import co.cask.cdap.common.NotFoundException;
 import co.cask.cdap.common.entity.EntityExistenceVerifier;
+import co.cask.cdap.common.namespace.NamespaceQueryAdmin;
 import co.cask.cdap.proto.id.NamespaceId;
-import co.cask.cdap.store.NamespaceStore;
+import com.google.common.base.Throwables;
 import com.google.inject.Inject;
 
 /**
  * {@link EntityExistenceVerifier} for {@link NamespaceId namespaces}.
  */
 public class NamespaceExistenceVerifier implements EntityExistenceVerifier<NamespaceId> {
 
-  private final NamespaceStore nsStore;
+  private final NamespaceQueryAdmin namespaceQueryAdmin;
 
   @Inject
-  NamespaceExistenceVerifier(NamespaceStore nsStore) {
-    this.nsStore = nsStore;
+  NamespaceExistenceVerifier(NamespaceQueryAdmin namespaceQueryAdmin) {
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
   }
 
 
   @Override
   public void ensureExists(NamespaceId namespaceId) throws NotFoundException {
     if (!NamespaceId.SYSTEM.equals(namespaceId)) {
-      if (nsStore.get(namespaceId.toId()) == null) {
-        throw new NamespaceNotFoundException(namespaceId.toId());
+      try {
+        if (namespaceQueryAdmin.get(namespaceId.toId()) == null) {
```

Comment:
@bdmogal I am changing this to call exists rather than get which has authorization. ensuring SYSTEM Namespace should not need authorization. Let me know if you have a different opinion.